src/custom/components/Popups/PopupItemMod.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useCallback, useContext, useEffect } from 'react'
+import { X } from 'react-feather'
+import { useSpring } from 'react-spring/web'
+import styled, { ThemeContext } from 'styled-components'
+import { animated } from 'react-spring'
+import { PopupContent } from 'state/application/actions'
+import { useRemovePopup } from 'state/application/hooks'
+import ListUpdatePopup from 'components/Popups/ListUpdatePopup'
+import TransactionPopup from './TransactionPopupMod'
+
+export const StyledClose = styled(X)`
+  position: absolute;
+  right: 10px;
+  top: 10px;
+
+  :hover {
+    cursor: pointer;
+  }
+`
+export const Popup = styled.div`
+  display: inline-block;
+  width: 100%;
+  padding: 1em;
+  background-color: ${({ theme }) => theme.bg1};
+  position: relative;
+  border-radius: 10px;
+  padding: 20px;
+  padding-right: 35px;
+  overflow: hidden;
+
+  ${({ theme }) => theme.mediaWidth.upToSmall`
+    min-width: 290px;
+    &:not(:last-of-type) {
+      margin-right: 20px;
+    }
+  `}
+`
+export const Fader = styled.div`
+  position: absolute;
+  bottom: 0px;
+  left: 0px;
+  width: 100%;
+  height: 2px;
+  background-color: ${({ theme }) => theme.bg3};
+`
+
+const AnimatedFader = animated(Fader)
+
+export default function PopupItem({
+  removeAfterMs,
+  content,
+  popKey,
+  PopupCustom
+}: {
+  removeAfterMs: number | null
+  content: PopupContent
+  popKey: string
+  PopupCustom: React.ElementType
+}) {
+  const removePopup = useRemovePopup()
+  const removeThisPopup = useCallback(() => removePopup(popKey), [popKey, removePopup])
+  useEffect(() => {
+    if (removeAfterMs === null) return undefined
+
+    const timeout = setTimeout(() => {
+      removeThisPopup()
+    }, removeAfterMs)
+
+    return () => {
+      clearTimeout(timeout)
+    }
+  }, [removeAfterMs, removeThisPopup])
+
+  const theme = useContext(ThemeContext)
+
+  let popupContent
+  if ('txn' in content) {
+    const {
+      txn: { hash, success, summary }
+    } = content
+    popupContent = <TransactionPopup hash={hash} success={success} summary={summary} />
+  } else if ('listUpdate' in content) {
+    const {
+      listUpdate: { listUrl, oldList, newList, auto }
+    } = content
+    popupContent = <ListUpdatePopup popKey={popKey} listUrl={listUrl} oldList={oldList} newList={newList} auto={auto} />
+  } else if ('metatxn' in content) {
+    const {
+      metatxn: { id, success, summary }
+    } = content
+    popupContent = <TransactionPopup hash={id} success={success} summary={summary} />
+  }
+
+  const faderStyle = useSpring({
+    from: { width: '100%' },
+    to: { width: '0%' },
+    config: { duration: removeAfterMs ?? undefined }
+  })
+
+  return (
+    <PopupCustom>
```

Comment:
you don't need to pass the popup custom afak

here it was returning the styled component. 
I think you can extend it from the `PopupIten.tsx` directly. There's only one gotcha, you need to ass a className in the parameters (its a requirement in styled components if you want to extend them) 

Use `MenuMod` as a reference, notice the `className={props?.className}`  and the 

```ts
export const StyledMenu = styled(MenuMod)`
  hr {
    margin: 15px 0;
  }
`
```

Reply:
@anxolin Refactored it now to pass the className from child -> parent (index.tsx). Let me know your thoughts.